dist/common/scripts/scylla_util.py

```diff
@@ -106,6 +110,135 @@ def curl(url, byte=False):
                 raise
 
 
+class gcp_instance:
+    """Describe several aspects of the current GCP instance"""
+
+    from psutil import virtual_memory
+    from psutil import cpu_count
+
+    def is_gce_instance(self):
+      """Check if it's GCE instance via DNS lookup to metadata server.
+      """
+      import socket
+      try:
+        addrlist = socket.getaddrinfo('metadata.google.internal', 80)
+      except socket.gaierror:
+        return False
+      for res in addrlist:
+        af, socktype, proto, canonname, sa = res
+        if (af == socket.AF_INET):
+          addr, port = sa
+          if (addr == "169.254.169.254"):
+            return True
+      return False
+
+    def __memory_size_gb(self):
+        mem = psutil.virtual_memory()
+        return mem.total/1024/1024/1024
+
+    def __cpu_count(self):
+        return psutil.cpu_count()
+
+    def __instance_metadata(self, path):
+        return curl("http://169.254.169.254/computeMetadata/v1/instance/" + path+"?recursive=true", headers="Metadata-Flavor: Google")
+        #169.254.169.254 is metadata.google.internal
+
+    def __disk_name(self, dev):
+        name = re.compile(r"(?:/dev/)?(?P<devname>[a-zA-Z]+)\d*")
+        return name.search(dev).group("devname")
+
+    def __device_exists(self, dev):
+        if dev[0:4] != "/dev":
+            dev = "/dev/%s" % dev
+        return os.path.exists(dev)
+
+    def get_blockdev_size(self, blockdevname):
+      "Get the file size by checking sysfs"
+      blocks = int(open('/sys/block/{blockdevname}/size'.format(**locals())).read())
+      block_size_bytes = int(open('/sys/block/{blockdevname}/queue/physical_block_size'.format(**locals())).read())
+      return blocks * block_size_bytes
```

Comment:
The most straight forward is to query the Google metadata server:

```
[vladz@vladz-centos7-nvme-check ~]$ curl "http://metadata.google.internal/computeMetadata/v1/instance/disks/1/" -H "Metadata-Flavor: Google"
device-name
index
interface
mode
type
[vladz@vladz-centos7-nvme-check ~]$ curl "http://metadata.google.internal/computeMetadata/v1/instance/disks/1/device-name" -H "Metadata-Flavor: Google"
local-ssd-0[vladz@vladz-centos7-nvme-check ~]$ curl "http://metadata.google.internal/computeMetadata/v1/instance/disks/1/interface" -H "Metadata-Flavor: Google"
NVME[vladz@vladz-centos7-nvme-check ~]$ curl "http://metadata.google.internal/computeMetadata/v1/instance/disks/1/type" -H "Metadata-Flavor: Google"
LOCAL-SSD[vladz@vladz-centos7-nvme-check ~]$ 
```

Reply:
```
[vladz@vladz-centos7-nvme-check ~]$ curl "http://metadata.google.internal/computeMetadata/v1/instance/disks/" -H "Metadata-Flavor: Google"
0/
1/
2/
```

Reply:
ok, so this is a gotcha
I tested only with physical nvmes and it seems google partitions the 3TB nvme which backs up small disks and kernel is provided with parent info ... d'oh! (I'd even say this is googles virtualization bug of presenting the disk to VM)
so I need some better way

Reply:
I wanted to avoid parsing too much data from metadata server

Reply:
I understand and I agree that this is their bug, however we can't wait till they fix it.
IMO the best way to move forward is to work with their metadata server output.

Reply:
metadata doesn't contain the disk size,
previously I used the seek method, so that would be a portable check (I think there is no spinning on nvmes, I'd expect it'd jump directly till the end)
resp. on our talk we agreed that I will revert back to previous hardcode of the disk size to 375GB

Reply:
as agreed I will take the number of disks from metadata
then fetch the root device if it is nvme, if not, then I'd keep all the disks for the raid (as our raid setup does that), 
if yes, then I'd subtract this from the metadata count

Reply:
resolved with metadata check